GrandCentralBoard/Widgets/WebsiteAnalytics/WebsiteAnalyticsWidget.swift

```diff
@@ -9,44 +9,99 @@ import GCBCore
 
 
 final class WebsiteAnalyticsWidget: WidgetControlling {
+    private enum State {
+        case Loading
+        case Data(reports: [PageViewsRowReport])
+        case NoData
+        case Error
+    }
 
     private let widgetView: TableWidgetView
-    private let widgetViewWrapper: UIView
-    let sources: [UpdatingSource]
+    private let mainView: UIView
 
-    var view: UIView {
-        return widgetViewWrapper
+    private var state: State = .Loading {
+        didSet {
+            changeFromState(oldValue, toState: state)
+        }
     }
 
     init(sources: [UpdatingSource]) {
         self.widgetView = TableWidgetView.fromNib()
         self.sources = sources
 
-        let viewModel = WidgetTemplateViewModel(title: "BLOGPOSTS", subtitle: "MOST POPULAR", contentView: widgetView)
+        let viewModel = WidgetTemplateViewModel(title: "BLOGPOSTS".localized.uppercaseString,
+                                                subtitle: "MOST POPULAR".localized.uppercaseString,
+                                                contentView: widgetView)
         let layoutSettings = WidgetTemplateLayoutSettings(contentMargin: UIEdgeInsets(top: -90, left: 0, bottom: 0, right: 0))
-        widgetViewWrapper = WidgetTemplateView.viewWithViewModel(viewModel, layoutSettings: layoutSettings)
+        let widgetViewWithHeader = WidgetTemplateView.viewWithViewModel(viewModel, layoutSettings: layoutSettings)
+
+        mainView = UIView(frame: widgetViewWithHeader.frame)
+        mainView.fillViewWithView(widgetViewWithHeader, animated: false)
     }
 
+
+    // MARK: HavingSources
+
+    let sources: [UpdatingSource]
+
+
+    // MARK: WidgetControlling
+
+    var view: UIView {
+        return mainView
+    }
+
+
+    // MARK: Updateable
+
     func update(source: UpdatingSource) {
         switch source {
-        case let source as PageViewsSource:
-            fetchAnalyticsFromSource(source)
-        default:
-            assertionFailure("Expected `source` as instance of `GoogleAnalyticsSource`.")
+            case let source as PageViewsSource:
+                fetchAnalyticsFromSource(source)
+            default:
+                assertionFailure("Expected `source` as instance of `GoogleAnalyticsSource`.")
         }
     }
 
     private func fetchAnalyticsFromSource(source: PageViewsSource) {
         source.read { [weak self] result in
             switch result {
-            case .Success(let reports):
-                self?.updateWithReports(reports)
-            case .Failure:
-                break
+                case .Success(let reports):
```

Comment:
We had discussion at the beginning and decided to use standard Xcode indentation in `switch` statements